packages/components/src/components/hds/advanced-table/helpers.ts

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import type { HdsAdvancedTableTdSignature } from './td';
+import type { HdsAdvancedTableThSignature } from './th';
+
+const getAllFocusableChildren = (element: HTMLElement): NodeListOf<Element> => {
+  return element.querySelectorAll(
+    'button:not([disabled]), a[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), [tabindex]:not([tabindex="-1"]):not([disabled])'
+  );
+};
+
+const getParentCell = (element: HTMLElement): Element | null | undefined => {
+  const targetParent = element.parentElement;
+
+  if (targetParent) {
+    return targetParent.closest(
+      '.hds-advanced-table__th, .hds-advanced-table__td'
+    );
+  }
+};
+
+const handleGridCellChildKeyPress = (event: KeyboardEvent): void => {
+  const { key, target } = event;
+
+  if (target instanceof HTMLElement) {
+    const cell = getParentCell(target);
+    if (cell instanceof HTMLElement) {
+      const allCellFocusableChildren = Array.from(
+        getAllFocusableChildren(cell)
+      ) as HTMLElement[];
+
+      if (key === 'Tab') {
+        const indexOfTarget = allCellFocusableChildren.indexOf(target);
+        let newTarget;
+
+        if (indexOfTarget === 0 && event.shiftKey) {
+          event.preventDefault();
+          newTarget =
+            allCellFocusableChildren[allCellFocusableChildren.length - 1];
+        }
+
+        if (indexOfTarget === allCellFocusableChildren.length - 1) {
+          event.preventDefault();
+          newTarget = allCellFocusableChildren[0];
+        }
+
+        if (newTarget) {
+          newTarget.focus();
+        }
+      }
+
+      if (key === 'Escape') {
```

Comment:
@MelSumner another heads up, after looking into the options for the issue with escape closing both the tooltip and re-focusing the cell - there is no option that scales well here to try and prevent that. It would be very hacky (like making a new tooltip just for advanced table) and wouldn't work if a consumer put something in a cell that also has an escape key handler.

Leaving the behavior as is.